services/payments/src/simcore_service_payments/services/auth.py

```diff
@@ -0,0 +1,86 @@
+"""
+Implements OAuth2 with Password and Bearer (w/ JWT tokens)
+
+"""
+
+from datetime import timedelta
+
+import arrow
+from fastapi import HTTPException, status
+from jose import JWTError, jwt
+from servicelib.utils_secrets import compare_secrets
+
+from ..core.settings import ApplicationSettings
+from ..models.auth import SessionData
+
+
+def authenticate_user(username: str, password: str, settings: ApplicationSettings):
+    return compare_secrets(
+        username + password,
+        expected=settings.PAYMENTS_USERNAME
+        + settings.PAYMENTS_PASSWORD.get_secret_value(),
+    )
+
+
+#
+# JW Tokens
+#
+
+
+_ALGORITHM = "HS256"
+
+
+def encode_access_token(username: str, settings: ApplicationSettings) -> str:
+    expire = arrow.utcnow().datetime + timedelta(
+        minutes=settings.PAYMENTS_ACCESS_TOKEN_EXPIRE_MINUTES
+    )
+    # SEE https://jwt.io/introduction/
+    payload_claims = {
+        # Registered claims
+        "sub": username,  # subject: https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2
+        "exp": expire,  # expiration date: https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.4
+    }
+    json_web_token: str = jwt.encode(
+        payload_claims,
+        key=settings.PAYMENTS_ACCESS_TOKEN_SECRET_KEY.get_secret_value(),
+        algorithm=_ALGORITHM,
+    )
+    return json_web_token
+
+
+def decode_access_token(token: str, settings: ApplicationSettings) -> str | None:
+    """
+    Raises:
+        JWTError
+    """
+    claims = jwt.decode(
+        token,
+        settings.PAYMENTS_ACCESS_TOKEN_SECRET_KEY.get_secret_value(),
+        algorithms=[_ALGORITHM],
+    )
+    username: str | None = claims.get("sub", None)
```

Comment:
Q: do you not care about the exp field here?

Reply:
No, because `jwt.decode` already checks that and raises `ExpiredSignatureError`